### PR TITLE
Fix for PR #775. Disable xdlops for non gfx908

### PR DESCRIPTION
**Pull request by shurale-nkn**
Fixes bug in tests (partially unblocks #944)

Stage: FP32 OpenCL install all vega 20
and other non gfx908
Failed tests:
test_conv_igemm_dynamic_xdlops_fwd
test_conv_igemm_dynamic_xdlops_bwd
test_conv_igemm_dynamic_xdlops_wrw
Convolution cannot be executed due to incorrect params

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -228,52 +228,83 @@ set_tests_properties(test_sqlite_perfdb test_perfdb
...
+    set(is_half_check)
+    bool_and_f(${MIOPEN_TEST_HALF} ${PARSE_ALLOW_HALF} half_check)
+    
+    set(is_bfloat16_check)
+    bool_and_f(${MIOPEN_TEST_BFLOAT16} ${PARSE_ALLOW_BFLOAT16} bfloat16_check)
+
+    set(is_int8_check)
+    bool_and_f(${MIOPEN_TEST_INT8} ${PARSE_ALLOW_INT8} int8_check)
+    
+    set(is_miotensile_chech)
```
Comment (atamazov) [inline test/CMakeLists.txt:287]:
```suggestion
    set(is_miotensile_check)
```

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
Merge conflicts

**Comment by atamazov** [comment]
@shurale-nkn Basically, this looks good. Let's have a quick call tomorrow.

**Comment by shurale-nkn** [comment]
First run passed CI

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -54,9 +55,8 @@ find_package(Threads REQUIRED)
 add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} --output-on-failure -C ${CMAKE_CFG_INTDIR})
 add_custom_target(tests)
 
-set(MIOPEN_TEST_NONXDLOPS TRUE)
 if(MIOPEN_TEST_GFX908)
-    set(MIOPEN_TEST_NONXDLOPS FALSE)
+    set(MIOPEN_TEST_GFX906 FALSE)
```
Comment (atamazov) [inline test/CMakeLists.txt:59]: Many tests can run on gfx900. Please rename GFX906 to VEGA everywhere.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -228,52 +228,121 @@ set_tests_properties(test_sqlite_perfdb test_perfdb
...
+    bool_or_f(${is_full_check} ${MIOPEN_TEST_ALL} is_full_check)
+
+
+    add_custom_target(${NAME} ${PARSE_UNPARSED_ARGUMENTS})
+    add_test(NAME ${NAME} COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR} --target ${NAME})
+    if(  (is_gfx906_check OR is_gfx908_check)
+     AND (is_miotensile_check AND is_full_check)
+     AND ( is_half_check OR is_bfloat16_check OR is_int8_check OR MIOPEN_TEST_FLOAT)
+    )
+        if(PARSE_PERF_TEST)
```
Comment (atamazov) [inline test/CMakeLists.txt:340]: Please replace PERF_TEST with TEST_PERF_DB_RECORD_NOT_FOUND (yes, this looks terrible, so please feel free to invent something better, if you can)))

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
@shurale-nkn We'll need to explain the new features introduced in this PR -- the values of new attributes and their meanings, for example. Can you please prepare some internal documentation? Let's have a quick call tomorrow and discuss. Thanks.

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -34,7 +34,7 @@ option( MIOPEN_TEST_HALF "Test in half mode" OFF )
 option( MIOPEN_TEST_INT8 "Test in int8 mode" OFF )
 option( MIOPEN_TEST_BFLOAT16 "Test in bfloat16 mode" OFF )
 option( MIOPEN_TEST_GFX908 "Test arch gfx908 mode" OFF )
-option( MIOPEN_TEST_GFX906 "Test arch gfx906 mode" ON )
+option( MIOPEN_TEST_VEGA "Test arch VEGA mode" ON )
```
Comment (atamazov) [inline test/CMakeLists.txt:37]: [Recommendation] 
```suggestion
option( MIOPEN_TEST_VEGA "Test arch VEGA (gfx900, gfx906)" ON )
```

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -279,13 +279,35 @@ function(option_support_check is_anabled is_disabled default_result result)
...
+# You can use any number of options, provided that options do not conflict 
+#   (e.g. "HALF_ENABLE HALF DISABLE" is illegal)
+# 1)First describes supported data type. ( HALF BF16 INT8 FLOAT ...)
+#   The option can be enabled or disabled by using '_ENABLED' and '_DISABLED' suffix.
+#   If nothing is specified, the default value is taken. 
+#       Default: HALF=disabled, BF16=disabled, INT8=disabled, FLOAT=enabled.
+# 2)Second options type describes support GPU types (gfx900, gfx906, gfx908 ...)
+#   The option can be enabled or disabled by using '_ENABLED' and '_DISABLED' suffix.
+#   If nothing is specified, the default value is taken.
+#       Default: (gfx900&gfx906)=VEGA=enabled, GFX908=disabled.
```
Comment (atamazov) [inline test/CMakeLists.txt:294]:
```suggestion
#       Default: VEGA=enabled, GFX908=disabled.
```

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -279,13 +279,35 @@ function(option_support_check is_anabled is_disabled default_result result)
     endif()
 endfunction()
 
+# The add_custom_test function contains options to describe the conditions,
+# under which new custom_tests should be run. Options are divided into several types. 
+# The option can be enabled or disabled, if nothing is specified, the default value is taken.
+# You can use any number of options, provided that options do not conflict 
+#   (e.g. "HALF_ENABLE HALF DISABLE" is illegal)
```
Comment (atamazov) [inline test/CMakeLists.txt:286]:
```suggestion
#   (e.g. "HALF_ENABLE HALF_DISABLE" is illegal)
```

**Review by atamazov — APPROVED** [review]
LGTM!

**Pull request merged**